driver-sync/src/test/functional/com/mongodb/client/CrudProseTest.java

```diff
@@ -16,181 +16,411 @@
+            assertEquals(1, secondEvent.getCommand().getArray("ops").size());
+            assertEquals(firstEvent.getOperationId(), secondEvent.getOperationId());
         }
+    }
 
-        try {
-            collection.insertMany(asList(new Document("x", 1)));
-            fail("Should throw, as document doesn't match schema");
-        } catch (MongoBulkWriteException e) {
-            // These assertions doesn't do exactly what's required by the specification, but it's simpler to implement and nearly as
-            // effective
-            assertTrue(e.getMessage().contains("Write errors"));
-            assertEquals(1, e.getWriteErrors().size());
-            if (serverVersionAtLeast(5, 0)) {
-                assertFalse(e.getWriteErrors().get(0).getDetails().isEmpty());
+    @DisplayName("8. MongoClient.bulkWrite handles a cursor requiring getMore within a transaction")
+    @Test
+    void testBulkWriteHandlesCursorRequiringGetMoreWithinTransaction() {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement prose test https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.md#8-mongoclientbulkwrite-handles-a-cursor-requiring-getmore-within-a-transaction");
+    }
+
+    @DisplayName("10. MongoClient.bulkWrite returns error for unacknowledged too-large insert")
+    @ParameterizedTest
+    @ValueSource(strings = {"insert", "replace"})
+    void testBulkWriteErrorsForUnacknowledgedTooLargeInsert(final String operationType) {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement maxBsonObjectSize validation https://jira.mongodb.org/browse/JAVA-5529");
+        try (MongoClient client = createMongoClient(getMongoClientSettingsBuilder()
+                .writeConcern(WriteConcern.ACKNOWLEDGED))) {
+            int maxBsonObjectSize = droppedDatabase(client).runCommand(new Document("hello", 1)).getInteger("maxBsonObjectSize");
+            Document document = new Document("a", join("", nCopies(maxBsonObjectSize, "b")));
+            ClientNamespacedWriteModel model;
+            switch (operationType) {
+                case "insert": {
+                    model = ClientNamespacedWriteModel.insertOne(namespace(), document);
+                    break;
+                }
+                case "replace": {
+                    model = ClientNamespacedWriteModel.replaceOne(namespace(), Filters.empty(), document);
+                    break;
+                }
+                default: {
+                    throw Assertions.fail(operationType);
+                }
             }
+            assertThrows(MongoClientException.class, () -> client.bulkWrite(singletonList(model)));
+        }
+    }
```

Comment:
I realized this test can't pass for now only after implementing it. Decided to leave the implementation, though the test is skipped.